src/engine/filters.ts

```diff
@@ -0,0 +1,12 @@
+import { vectorDistance } from '../utils';
```

Comment:
vectorDistance should be called dotProduct, it is the mathematical operation on two vectors.